server/src/gnome/prompter.rs

```diff
@@ -0,0 +1,205 @@
+// org.gnome.keyring.Prompter
+// https://gitlab.gnome.org/GNOME/gcr/-/blob/main/gcr/org.gnome.keyring.Prompter.xml
+
+use std::collections::HashMap;
+
+use clap::error::Result;
+use oo7::dbus::ServiceError;
+use zbus::{
+    interface, proxy, zvariant,
+    zvariant::{OwnedObjectPath, OwnedValue, Value},
+};
+
+use crate::{
+    gnome::secret_exchange::SecretExchange,
+    prompt::{Prompt, PromptRole},
+    service::Service,
+};
+
+// org.gnome.keyring.internal.Prompter
+
+#[proxy(
+    default_service = "org.gnome.keyring.SystemPrompter",
+    interface = "org.gnome.keyring.internal.Prompter",
+    default_path = "/org/gnome/keyring/Prompter"
+)]
+pub trait Prompter {
+    fn begin_prompting(&self, callback: &OwnedObjectPath) -> Result<(), ServiceError>;
+
+    fn perform_prompt(
+        &self,
+        callback: OwnedObjectPath,
+        type_: &str,
+        properties: HashMap<&str, OwnedValue>,
```

Comment:
Type safety, like mentioned before.

Reply:
Done, thanks!
Quick question: why using a HashMaps isn't type safe? I couldn't find an answer for this from the zbus book.

Reply:
What i meant here is that you can pass an arbitrary dict when in reality it is a struct but as the only way to pass that over dbus is to use a dict, we still shouldn't use a dict from Rust.

Reply:
Understood, thanks and sorry for the delay.